src/utils/got/got.ts

```diff
@@ -0,0 +1,90 @@
+enum METHOD {
+    GET = 'GET',
+    POST = 'POST',
+    PUT = 'PUT',
+    PATCH = 'PATCH',
+    DELETE = 'DELETE',
+}
+
+type Options = {
+    method?: string;
+    data?: any;
+    timeout?: number;
+    headers?: Record<string, string>;
+};
+
+function queryStringify(data: any) {
+    if (typeof data !== 'object') {
+        throw new Error('Data must be object');
+    }
+    let out = '?';
+    Object.keys(data).forEach((key) => {
+        out += `${key}=${data[key]}&`;
+    });
+    out = out.slice(0, -1);
+    return out.replace(/ /g, '%20');
```

Comment:
Решение через мутацию выглядит довольно сложным и дебажить это будет затруднительно. Я предложу более функциональное решение через комбинацию функций
```suggestion
       const stringified = Object
        .entries(data)
        .map(([key, value]) => `${key}=${value}`)
        .join('&')

    return '?' + stringified
```